sdk/storage/src/table/clients/entity_client.rs

```diff
@@ -0,0 +1,94 @@
+use crate::table::prelude::*;
+use crate::table::requests::*;
+use azure_core::errors::AzureError;
+use bytes::Bytes;
+use http::method::Method;
+use http::request::{Builder, Request};
+use std::sync::Arc;
+use url::Url;
+
+pub trait AsEntityClient<RK: Into<String>> {
+    fn as_entity_client(&self, row_key: RK) -> Result<Arc<EntityClient>, url::ParseError>;
+}
+
+impl<RK: Into<String>> AsEntityClient<RK> for Arc<PartitionKeyClient> {
+    fn as_entity_client(&self, row_key: RK) -> Result<Arc<EntityClient>, url::ParseError> {
+        EntityClient::new(self.clone(), row_key)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct EntityClient {
+    partition_key_client: Arc<PartitionKeyClient>,
+    row_key: String,
+    url: Url,
+}
+
+impl EntityClient {
+    pub(crate) fn new<RK: Into<String>>(
+        partition_key_client: Arc<PartitionKeyClient>,
+        row_key: RK,
+    ) -> Result<Arc<Self>, url::ParseError> {
```

Comment:
Still not convinced that returning an `Arc` is the right thing. If we expect users to close the entity client, we should just make it cheap to clone.